master/security/high-connection-workloads.md

```diff
@@ -0,0 +1,93 @@
+
+#### Calico doNotTrack network policy
+
+The Calico global network policy option, **doNotTrack**, indicates to apply the rules in the policy before connection tracking, and that packets allowed by these rules should not be tracked. The policy is applied early in the Linux packet processing pipeline, before any regular network policy rules, and independent of the policy order field. 
+
+Unlike normal network policy rules, doNotTrack network policy rules are stateless, meaning you must explicitly specify rules to allow return traffic that would normally be automatically allowed by conntrack. For example, for a server on port 999, the policy must include an ingress rule allowing inbound traffic to port 999, and an egress rule to allow outbound traffic from port 999. 
+
+In a doNotTrack policy:
+- Ingress rules apply to all incoming traffic through a host endpoint, regardless of where the traffic is going 
+- Egress rules apply only to traffic that is sent from the host endpoint (not a local workload)
+
+Finally, you must add an **applyOnForward: true expression** for a **doNotTrack policy** to work.
+
+### Before you begin...
+
+Before creating a **doNotTrack** network policy, read this [blog](https://www.tigera.io/blog/when-linux-conntrack-is-no-longer-your-friend/) to understand use cases, benefits, and trade offs. 
+
+### How to
+
+#### Bypass connection traffic for high connection server
+
+In the following example, a memcached server pod with **hostNetwork: true** was scheduled on the node memcached-node-1. We create a HostEndpoint for the node. Next, we create a GlobalNetwork Policy with symmetrical rules for ingress and egress with doNotTrack and applyOnForward set to true.
+
+```
+apiVersion: projectcalico.org/v3
+kind: HostEndpoint
+metadata:
+  name: memcached-node-1-eth0
+  labels:
+    memcached: server
+spec:
+  interfaceName: eth0  
+  node: memcached-node-1  
+  expectedIPs:
+    - 10.128.0.162  
+---
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: memcached-server
+spec:
+  selector: memcached == 'server'
+  applyOnForward: true
+  doNotTrack: true
+  ingress:
+    - action: Allow
+      protocol: TCP
+      source:
+        selector: memcached == 'client'
+      destination:
+        ports:
+          - 12211
+  egress:
+    - action: Allow
+      protocol: TCP
+      source:
+        ports:
+          - 12211
+      destination:
+        selector: memcached == 'client'
+```
+### Above and beyond
+
+- [Calico GlobalNetworkPolicy](https://docs.projectcalico.org/v3.6/reference/calicoctl/resources/globalnetworkpolicy)
+- [Blog: When Linux conntrack is no longer your friend](https://www.tigera.io/blog/when-linux-conntrack-is-no-longer-your-friend/)
```

Comment:
Does it make sense for this blog to be both "Before you begin" as well as "Above and beyond"?

Reply:
It doesn't hurt to have both places, but I removed the one at the end.